src/state/slices/tradeInputSlice/selectors.ts

```diff
@@ -99,9 +101,32 @@ export const selectFirstHopSellAccountId = createSelector(
   },
 )
 
-// selects the account ID we're selling from for the other hops
-// for posterity, every hop always sells from the same account as the first hop
-export const selectSecondHopSellAccountId = selectFirstHopSellAccountId
+// if multi-hop, selects the account ID we're selling from for fee asset of the last hop and its account number
+// else, selects the AccountId from the first hop
+export const selectSecondHopSellAccountId = createSelector(
+  selectAccountNumberParamFromFilter,
+  selectAccountIdByAccountNumberAndChainId,
+  selectChainIdParamFromFilter,
+  // selectIsActiveQuoteMultiHop,
+  selectFirstHopSellAccountId,
+  (
+    sellAssetAccountNumber,
+    sellAssetAccountIds,
+    sellAssetChainId,
+    // isMultiHopTrade,
+    firstHopSellAccountId,
+  ) => {
+    // TODO(gomes): fix circular deps here
+    const isMultiHopTrade = true
+    if (!isMultiHopTrade) return firstHopSellAccountId
```

Comment:
We probably want to implement `selectLastHopSellAccountId` properly instead, and wire it up since it isn't used yet

Reply:
Actually we don't